eventhubsprocessor/partition_manager.py

```diff
@@ -0,0 +1,280 @@
+"""
+Author: Aaron (Ari) Bornstien
+"""
+import time
+import logging
+import asyncio
+import concurrent.futures
+from collections import Counter
+from bs4 import BeautifulSoup
+import requests
+from eventhubsprocessor.eh_partition_pump import EventHubPartitionPump
+from eventhubsprocessor.cancellation_token import CancellationToken
+
+class PartitionManager:
+    """
+    Manages the partition event pump execution
+    """
+    def __init__(self, host):
+        self.host = host
+        self.partition_pumps = {}
+        self.partition_ids = None
+        self.run_task = None
+        self.pump_executor = None
+        self.cancellation_token = CancellationToken()
+
+    async def get_partition_ids_async(self):
+        """
+        Returns a list of all the event hub partition ids
+        """
+        if not self.partition_ids:
+            try:
+                headers = {"Content-Type":"application/json;type=entry;charset=utf-8",
+                           "Authorization": self.host.eh_rest_auth["token"],
+                           "Host":"k8sworkshop.servicebus.windows.net"
```

Comment:
Set Host from connection address

Reply:
Good catch I originally tested this function and then abstracted it i must have missed this one.